packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -1392,6 +1431,15 @@ export class SamplerOperations {
                             ...intermediateTokens.map(t => [takerToken, t, makerToken]),
                         ].map(path => this.getUniswapV3SellQuotes(router, quoter, path, takerFillAmounts));
                     }
+                    case ERC20BridgeSource.Lido: {
+                        const lidoInfo = LIDO_INFO_BY_CHAIN[this.chainId];
+
+                        if (lidoInfo.stEthToken === NULL_ADDRESS) {
+                            return [];
```

Comment:
I would check that takerToken == WETH and makerToken == stETH here. That way we can avoid hitting the sampler whenever these tokens aren't involved. Similar to Curve, where we only check certain curves if they involve those tokens.